cmd/signature-v4.go

```diff
@@ -248,6 +248,12 @@ func doesPresignedSignatureMatch(hashedPayload string, r *http.Request, region s
 
 	// Save other headers available in the request parameters.
 	for k, v := range req.URL.Query() {
+
+		// Handle the metadata in presigned put query string
+		if strings.Contains(k, "x-amz-meta-") {
```

Comment:
Hey @harshavardhana !
S3's presigned URL parameters are lower cased , They invalidate the signature if we send them a camel cased query parameter. 
So in our case should we handle it ?

Reply:
The problem is that aws-sdk-php is using a non-canonical form and that is a bug, but since we need to perhaps handle this also on our end. We can perhaps sign one separated with lower case and another with upper case just for `x-amz-meta` alone.

Reply:
Hi @harshavardhana  , The s3 validates the signature only if the `x-amz-meta` is lowercased in the query strings . ie, it signs with lowercased `x-amz-meta` . So the above change will overcome the behaviour.